src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -30,119 +30,85 @@
-     *
-     * @param key the key of the registry to reference. May come from another DeferredRegister through {@link #getRegistryKey()}.
-     * @param modid the namespace for all objects registered to this DeferredRegister
-     * @see #createOptional(ResourceKey, String)
-     * @see #create(IForgeRegistry, String)
-     * @see #create(ResourceLocation, String)
+     * @deprecated Use {@link #create(ResourceKey, String)} via {@link IForgeRegistry#getRegistryKey()}
      */
-    public static <B> DeferredRegister<B> create(ResourceKey<? extends Registry<B>> key, String modid)
+    @Deprecated(since = "1.20.1", forRemoval = true)
+    public static <B> DeferredRegister<B> create(IForgeRegistry<B> reg, String modid)
     {
-        return new DeferredRegister<>(key, modid, false);
+        return create(reg.getRegistryKey(), modid);
     }
 
     /**
-     * DeferredRegister factory for the optional existence of custom forge registries
-     * or {@link BuiltInRegistries vanilla registries} to lookup based on the provided registry key.
-     * Supports both registries that already exist or do not exist yet.
-     * <p>
-     * If the registry is never created, any {@link RegistryObject}s made from this DeferredRegister will never be filled but will not throw an exception.
-     *
-     * @param key the key of the registry to reference
-     * @param modid the namespace for all objects registered to this DeferredRegister
-     * @see #create(ResourceKey, String)
-     * @see #create(IForgeRegistry, String)
-     * @see #create(ResourceLocation, String)
+     * @deprecated Use {@link #create(ResourceKey, String)}
      */
+    @Deprecated(since = "1.20.1", forRemoval = true)
     public static <B> DeferredRegister<B> createOptional(ResourceKey<? extends Registry<B>> key, String modid)
     {
-        return new DeferredRegister<>(key, modid, true);
+        return create(key, modid);
     }
 
     /**
-     * DeferredRegister factory for custom forge registries or {@link BuiltInRegistries vanilla registries}
-     * to lookup based on the provided registry name. Supports both registries that already exist or do not exist yet.
-     * <p>
-     * If the registry is never created, any {@link RegistryObject}s made from this DeferredRegister will throw an exception.
-     * To allow the optional existence of a registry without error, use {@link #createOptional(ResourceLocation, String)}.
-     *
-     * @param registryName The name of the registry, should include namespace. May come from another DeferredRegister through {@link #getRegistryName()}.
-     * @param modid The namespace for all objects registered to this DeferredRegister
-     * @see #createOptional(ResourceLocation, String)
-     * @see #create(IForgeRegistry, String)
+     * Variant of {@link #create(ResourceKey, String)} that accepts a ResourceLocation instead of a ResourceKey.
```

Comment:
Personally, I think it is easier when the javadocs are duplicated for similar methods rather than requiring someone to read the javadocs elsewhere for this variant method.